src/db/handlers/mod.rs

```diff
@@ -0,0 +1,22 @@
+use crate::db::messages;
+use crate::db::models;
+use crate::db::DbExecutor;
+use crate::schema;
+use actix_web::actix::*;
+use actix_web::error;
+use diesel;
+use diesel::prelude::*;
+use diesel::r2d2::{ConnectionManager, Pool};
+
+impl Handler<messages::GetUser> for DbExecutor {
+    type Result = Result<Option<models::User>, actix_web::Error>;
+    fn handle(&mut self, msg: messages::GetUser, ctx: &mut Self::Context) -> Self::Result {
+        use crate::schema::users::dsl::*;
+        let conn = &self.0.get().unwrap();
```

Comment:
What happens if a connection doesn't exist in the pool? Is this even possible?

Reply:
This would happen if pool does not get created correctly. The worst thing that would happen is we panic in the middle of a request. 

It might be better to map the error type and then throw it the failure result with `?` if it fails. That way we respond with a more informative http response.